### ISSUE TITLE
`qiskit.transpile` with `optimization_level=3` does not optimize away an identity operation like `cp` with phase 0

### ISSUE THREAD

**Issue by dixr** [Qiskit/qiskit#13344]
### Environment

- **Qiskit version**: 1.2.4
- **Python version**: 3.11.3
- **Operating system**: Ubuntu Linux

### What is happening?

Why does qiskit.transpile() function with optimization level 3 not remove a cp (controlled phase) gate when its parameter is 0? Since a cp(0) gate is equivalent to an identity operation, it should be optimized away.

### How can we reproduce the issue?

```python
import qiskit
import qiskit_aer
circuit = qiskit.QuantumCircuit(2)
circuit.cp(0, 1, 0)
circuit = qiskit.transpile(circuit, qiskit_aer.AerSimulator(), optimization_level=3)
circuit.data
```
Output: 
```
[CircuitInstruction(operation=Instruction(name='cp', num_qubits=2, num_clbits=0, params=[0.0]), qubits=(Qubit(QuantumRegister(2, 'q'), 1), Qubit(QuantumRegister(2, 'q'), 0)), clbits=())]
```

### What should happen?

The gate should be removed.

**Comment by Cryoris** [comment]
**Short answer:** Yeah we currently don't do value-based optimization for gates that are already in the target basis (note that the `AerSimulator` accepts any standard gate in its basis). To resolve this you could set a specific basis that forces Qiskit to re-synthesize the operation. For example
```
import qiskit
circuit = qiskit.QuantumCircuit(2)
circuit.cp(0, 1, 0)
circuit = qiskit.transpile(circuit, basis_gates=["u", "cx"], optimization_level=3)  
```
That should also be in the correct format for the `AerSimulator`, I think.


**Longer answer:** This is generally an optimization we miss, here's a more basic example:
```
import qiskit

circuit = qiskit.QuantumCircuit(1)
circuit.ry(0, 0)
circuit = qiskit.transpile(circuit, basis_gates=["ry"], optimization_level=3, callback=callback)
print(circuit)  # Ry(0) present!

circuit = qiskit.transpile(circuit, basis_gates=["u"], optimization_level=3, callback=callback)
print(circuit)  # empty!
```
In the second example, the `Ry(0)` gets converted to `U(0,0,0)` and subsequently `Optimize1qGatesDecomposition` removes the gate. We could maybe expand that workflow to remove 0-angle rotations with the same tolerance that it uses in converting Euler angles to the identity 🤔

**Comment by Cryoris** [comment]
Update: this should get fixed by #12384.
